### PR TITLE
fix: Save new version of the workflow instead of the previous (no-changelog)

### PR DESCRIPTION
**Pull request by valya**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Review by krynble — COMMENTED** [review]
We might have missed it in the previous PR, but can we add the tests for those? Just to make sure that the function is being called?

File: packages/cli/test/integration/publicApi/workflows.test.ts [hunk 1]
```diff
@@ -679,6 +695,90 @@ describe('POST /workflows', () => {
 		expect(sharedWorkflow?.role).toEqual(workflowOwnerRole);
 	});
 
+	test('should create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/publicApi/workflows.test.ts:698]:
```suggestion
	test('should create workflow history version when licensed', async () => {
```

File: packages/cli/test/integration/publicApi/workflows.test.ts [hunk 2]
```diff
@@ -679,6 +695,90 @@ describe('POST /workflows', () => {
...
+			where: {
+				workflowId: id,
+			},
+		});
+		expect(historyVersion).not.toBeNull();
+		expect(historyVersion!.connections).toEqual(payload.connections);
+		expect(historyVersion!.nodes).toEqual(payload.nodes);
+	});
+
+	test('should not create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/publicApi/workflows.test.ts:744]:
```suggestion
	test('should not create workflow history version when not licensed', async () => {
```

File: packages/cli/test/integration/publicApi/workflows.test.ts [hunk 3]
```diff
@@ -835,6 +935,108 @@ describe('PUT /workflows/:id', () => {
 		);
 	});
 
+	test('should create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/publicApi/workflows.test.ts:938]:
```suggestion
	test('should create workflow history version when licensed', async () => {
```

File: packages/cli/test/integration/workflows.controller.test.ts [hunk 4]
```diff
@@ -46,6 +54,96 @@ describe('POST /workflows', () => {
 		const pinData = await testWithPinData(false);
 		expect(pinData).toBeNull();
 	});
+
+	test('should create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/workflows.controller.test.ts:58]:
```suggestion
	test('should create workflow history version when licensed', async () => {
```

File: packages/cli/test/integration/workflows.controller.test.ts [hunk 5]
```diff
@@ -46,6 +54,96 @@ describe('POST /workflows', () => {
...
+			where: {
+				workflowId: id,
+			},
+		});
+		expect(historyVersion).not.toBeNull();
+		expect(historyVersion!.connections).toEqual(payload.connections);
+		expect(historyVersion!.nodes).toEqual(payload.nodes);
+	});
+
+	test('should not create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/workflows.controller.test.ts:107]:
```suggestion
	test('should not create workflow history version when not licensed', async () => {
```

File: packages/cli/test/integration/workflows.controller.test.ts [hunk 6]
```diff
@@ -318,3 +416,111 @@ describe('GET /workflows', () => {
 		});
 	});
 });
+
+describe('PATCH /workflows/:id', () => {
+	test('should create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/workflows.controller.test.ts:421]:
```suggestion
	test('should create workflow history version when licensed', async () => {
```

File: packages/cli/test/integration/workflows.controller.test.ts [hunk 7]
```diff
@@ -318,3 +416,111 @@ describe('GET /workflows', () => {
...
+			where: {
+				workflowId: id,
+			},
+		});
+		expect(historyVersion).not.toBeNull();
+		expect(historyVersion!.connections).toEqual(payload.connections);
+		expect(historyVersion!.nodes).toEqual(payload.nodes);
+	});
+
+	test('should not create workflow history version', async () => {
```
Comment (krynble) [inline packages/cli/test/integration/workflows.controller.test.ts:478]:
```suggestion
	test('should not create workflow history version when not licensed', async () => {
```

**Review by krynble — DISMISSED** [review]
LGTM, just a few nitpics

**Review by krynble — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.12.2`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.12.2)

File: packages/cli/src/workflows/workflows.services.ts [hunk 8]
```diff
@@ -301,8 +301,8 @@ export class WorkflowsService {
 			);
 		}
 
-		if (isWorkflowHistoryLicensed()) {
-			await Container.get(WorkflowHistoryService).saveVersion(user, shared.workflow);
+		if (isWorkflowHistoryLicensed() && workflow.versionId !== shared.workflow.versionId) {
```
Comment (netroy) [inline packages/cli/src/workflows/workflows.services.ts:304]: @valya @krynble 
This change is breaking workflow activation/deactivation when `isWorkflowHistoryLicensed` returns true.

**Pull request merged**